src/jdk.javadoc/share/classes/jdk/javadoc/doclet/DocletEnvironment.java

```diff
@@ -136,7 +140,47 @@
      * @param type the type element
      * @return the file kind
      */
-    Kind getFileKind(TypeElement type);
+        Kind getFileKind(TypeElement type);
+
+        /**
+         * Returns a writer that can be used by a doclet to write non-diagnostic output,
+         * or {@code null} if no such writer is available.
+         *
+         * @apiNote
+         * The value may or may not be the same as that returned by {@link #getDiagnosticWriter()}.
+         *
+         * @implSpec
+         * This implementation returns {@code null}.
+         * The implementation provided by the `javadoc` tool to
+         * {@link Doclet#init(Locale, Reporter) initialize doclets}
+         * always returns a non-{@code null} value.
```

Comment:
Did you mean `{@link Doclet#run(DocletEnvironment) run doclets}` instead?

Reply:
Hmm, I guess I did. Good catch.